Libraries/LibWeb/CredentialManagement/CredentialsContainer.idl

```diff
@@ -0,0 +1,21 @@
+#import <CredentialManagement/Credential.idl>
+#import <CredentialManagement/FederatedCredential.idl>
+#import <CredentialManagement/PasswordCredential.idl>
+
+[Exposed=Window, SecureContext]
+interface CredentialsContainer {
+    Promise<Credential?> get(optional CredentialRequestOptions options = {});
+    Promise<undefined> store(Credential credential);
+    Promise<Credential?> create(optional CredentialCreationOptions options = {});
+    Promise<undefined> preventSilentAccess();
+};
+
+dictionary CredentialRequestOptions {
+    CredentialMediationRequirement mediation = "optional";
+    AbortSignal signal;
+};
+
+dictionary CredentialCreationOptions {
+    CredentialMediationRequirement mediation = "optional";
+    AbortSignal signal;
+};
```

Comment:
missing newline at end of file. please configure your editor to add these automagically.